### PR TITLE
BFD-3406: Snyk Upgrades

### PR DESCRIPTION
**Pull request by dmoorecode**
**JIRA Ticket:**
[BFD-3406](https://jira.cms.gov/browse/BFD-3406)

**User Story or Bug Summary:**
Update Black Versions and vulnerabilities for python code 

### What Does This PR Do?
Modify txt and .in configuration file to update packages 
ranging from encoding, formatting and security vulnerabilities 

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] The data dictionary has been updated with any field mapping changes, if any were made.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Review by mjburling — CHANGES_REQUESTED** [review]
This is shaping up, thanks!

Depending on the workload, specific libraries, and some other factors, updating the python versions could have unintended consequences. Can you be sure to run the compilation step with the version specified in the various requirements files, i.e. `requirements.txt`, `dev-requirements.txt`? Locally, you'll need to be able to support versions 3.9, 3.10, and 3.12, I believe.

Separately, we should look to upgrade and standardize on a recent version of python that works for our variety of use cases.

**Review by dondevun — APPROVED** [review]

**Review by JFU-NAVA-PBC — DISMISSED** [review]
LGTM

**Comment by dmoorecode** [comment]
Virtual environment installed for each python version needed. The appropriate packages and pip tools were compiled for the specified version for each modules to generate the correct txt files. All files submitted have been updated in pull request

File: ops/terraform/services/eft/lambda_src/sftp_outbound_transfer/requirements.in [hunk 1]
```diff
@@ -2,3 +2,4 @@ aws-lambda-powertools[all]
 paramiko>=3.4.0
 boto3>=1.34.5
 setuptools>=69.0.2
+black==24.3.0
```
[the remark this answers is not in the document]
Comment (dmoorecode) [reply ops/terraform/services/eft/lambda_src/sftp_outbound_transfer/requirements.in:5]: Yes you are correct, will correct encoding package to appear in dev only

File: ops/terraform/services/eft/modules/bfd_eft_outbound_o11y/lambda_src/outbound_slack_notifier/requirements.in [hunk 2]
```diff
@@ -1,2 +1,3 @@
 aws-lambda-powertools[all]
 setuptools>=69.0.2
+
```
Comment (mjburling) [inline ops/terraform/services/eft/modules/bfd_eft_outbound_o11y/lambda_src/outbound_slack_notifier/requirements.in:3]:
```suggestion
```

Comment (dmoorecode) [reply ops/terraform/services/eft/modules/bfd_eft_outbound_o11y/lambda_src/outbound_slack_notifier/requirements.in]: file modified removed /r/n from end-of-file

File: ops/terraform/services/eft/lambda_src/sftp_outbound_transfer/dev-requirements.txt [hunk 3]
```diff
@@ -2,85 +2,85 @@
 # This file is autogenerated by pip-compile with Python 3.12
 # by the following command:
 #
-#    pip-compile --allow-unsafe --constraint=requirements.txt --generate-hashes --output-file=dev-requirements.txt dev-requirements.in
+#    pip-compile --allow-unsafe --generate-hashes --output-file=dev-requirements.txt dev-requirements.in
```
Comment (mjburling) [inline ops/terraform/services/eft/lambda_src/sftp_outbound_transfer/dev-requirements.txt:5]: Apologies for the late addition–I just noticed the modest difference here and later in _this_ requirements file as a result. Could you re-institute the constraints flag and re-generate?
```suggestion
#    pip-compile --allow-unsafe --constraint=requirements.txt --generate-hashes --output-file=dev-requirements.txt dev-requirements.in
```

Comment (dmoorecode) [reply ops/terraform/services/eft/lambda_src/sftp_outbound_transfer/dev-requirements.txt]: complete

**Review by mjburling — APPROVED** [review]

**Review by dondevun — APPROVED** [review]

**Pull request merged**